### PR TITLE
Test hmnd

### PR DESCRIPTION
**Pull request by ManojJiSharma**

### CODE REVIEW

File: rosetta-client/src/wallet.rs [hunk 1]
```diff
@@ -105,6 +105,7 @@ impl Wallet {
         // self.client.finalized_block().await
         match &self.client {
             GenericClient::Astar(client) => client.finalized_block().await,
+            GenericClient::Humanode(client) => client.finalized_block().await,
             GenericClient::Ethereum(client) => client.finalized_block().await,
             GenericClient::Polkadot(client) => client.finalized_block().await,
         }
```
Comment (MOZGIII) [inline rosetta-client/src/wallet.rs:111]: Consider using https://docs.rs/trait_enum for things like this

File: chains/humanode/server/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+    async fn faucet(&self, address: &Address, value: u128) -> Result<Vec<u8>> {
+        // convert address
+        let dest = {
+            let address: H160 = address.address().parse()?;
+            let mut data = [0u8; 24];
+            data[0..4].copy_from_slice(b"evm:");
+            data[4..24].copy_from_slice(&address[..]);
+            let hash = sp_core::hashing::blake2_256(&data);
+            AccountId32::from(Into::<[u8; 32]>::into(hash))
+        };
```
Comment (MOZGIII) [inline chains/humanode/server/src/lib.rs:263]: I don't think this is valid for Humanode network

File: chains/humanode/server/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,503 @@
...
+            },
+            AddressFormat::Eip55 => {
+                // Frontier `eth_getBalance` returns the reducible_balance instead the free balance:
+                // https://github.com/paritytech/frontier/blob/polkadot-v0.9.43/frame/evm/src/lib.rs#L853-L855
+                // using substrate to get the free balance
+                let address = address
+                    .evm_to_ss58(Ss58AddressFormat::custom(42))
+                    .map_err(|err| anyhow::anyhow!("{}", err))?;
+                let account_info = self.account_info(&address, Some(block)).await?;
+                account_info.data.free
```
Comment (dmitrylavrenov) [inline chains/humanode/server/src/lib.rs:247]: Humanode version returns the exact free balance, we apply own patch to meet general eth logic - https://github.com/humanode-network/frontier/commit/a06c5f0d5db777efc9374480a23ac4ea0053f395.

Comment (MOZGIII) [reply chains/humanode/server/src/lib.rs:247]: Also, note that for the EVM integration you'd need to use the JSON-RPC API to interact with the network. Alternatively, you can use the `evm_system` pallet:
image
It is not possible though to submit an EVM transaction signing it with Substrate account signature.

**Comment by ManojJiSharma** [comment]
Need to reconfigure whole humanode

**Pull request closed without merging**